vendor/bat-native-ads/src/bat/ads/internal/ml/model/linear/linear_unittest.cc

```diff
@@ -0,0 +1,138 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <cmath>
+#include <vector>
+
+#include "bat/ads/internal/ml/data/vector_data.h"
+#include "bat/ads/internal/ml/model/linear/linear.h"
+
+#include "bat/ads/internal/json_helper.h"
+#include "bat/ads/internal/unittest_base.h"
+#include "bat/ads/internal/unittest_util.h"
+
+// npm run test -- brave_unit_tests --filter=BatAds*
+
+namespace ads {
+namespace ml {
+
+class BatAdsLinearModelTest : public UnitTestBase {
+ protected:
+  BatAdsLinearModelTest() = default;
+
+  ~BatAdsLinearModelTest() override = default;
+};
+
+TEST_F(BatAdsLinearModelTest, ThreeClassesPredictionTest) {
+  // Arrange
+  std::map<std::string, data::VectorData> weights = {
+      {"class_1", data::VectorData(std::vector<double>{1.0, 0.0, 0.0})},
+      {"class_2", data::VectorData(std::vector<double>{0.0, 1.0, 0.0})},
+      {"class_3", data::VectorData(std::vector<double>{0.0, 0.0, 1.0})}};
+
+  std::map<std::string, double> biases = {
```

Comment:
nit: const (check const correctness in other places also)

Reply:
Done